src/Map/ProspectorOverlayLayer.cs

```diff
@@ -379,6 +496,26 @@ public override void Dispose()
             base.Dispose();
         }
 
+        [HarmonyPatch(typeof(GuiDialogWorldMap), "TryClose")]
```

Comment:
This seems to work only when closing the map with escape, but not with the map key. Not sure what pressing M calls, but doesn't seem to be "TryClose"

Reply:
huh? weird
Works flawlessly for me for ESC, M, and closing via dialog closing cross. Hmm, maybe it's a server-side thing again as I was just testing in singleplayer...

Reply:
could be, i always test in multiplayer (joining is faster which makes it a nice side benefit)

https://github.com/anegostudios/vsessentialsmod/blob/e91826e75c3c03a7821c8b1e4944260a4e67dac2/Systems/WorldMap/GuiDialogWorldMap.cs#L15

It seems that the dialog is not really close but rather reopened as a different type. The OnMapOpenedClient and OnMapClosedClient don't seem to do what you would expect either.

Patching the WorldMapManager could work (ToggleMap) but when i tried it caused the mod to not load properly anymore.

I mean i could live with this bug (which is easy as i don't play anymore), but maybe you can figure out a workaround.

Reply:
Ahhh I got it now it was only a problem when minimap was enabled